sharding/indexHashedNodesCoordinatorWithRater.go

```diff
@@ -0,0 +1,142 @@
+package sharding
+
+import (
+	"fmt"
+	"github.com/ElrondNetwork/elrond-go/core"
+)
+
+type indexHashedNodesCoordinatorWithRater struct {
+	indexHashedNodesCoordinator
+	RatingReader
+}
+
+// NewIndexHashedNodesCoordinator creates a new index hashed group selector
+func NewIndexHashedNodesCoordinatorWithRater(arguments ArgNodesCoordinator, rater RatingReader) (*indexHashedNodesCoordinatorWithRater,
+	error) {
+	indexNodesCoordinator, err := NewIndexHashedNodesCoordinator(arguments)
```

Comment:
creating another object inside is generally done by factory. otherwise nodesCoordinator should be base class.

Reply:
fixed, comes as an argument